synce_dev.c

```diff
@@ -677,6 +681,91 @@ static int dev_step_dpll(struct synce_dev *dev)
 	return ret;
 }
 
+void synce_dev_get_ql(struct synce_dev *dev, uint8_t *ql)
```

Comment:
that exactly what is coded. if external source is best then return its ql

Reply:
what happens if there is no line-type interfaces configured?

Reply:
if no line interfaces? I think the all synce4l wont work :). it requires interfaces and up

Reply:
I mean the synce4l is allowed to run with only external source configured. It is a valid configuration, the old "external" mode which is achievable with config file that has no network interfaces configured.

Reply:
the old external mode was when you know that your QL from local SMA is best, and you **send** it to other devices using your configured port. if no ports, then how you synchronize the network?

Reply:
i.e. a user configures SMA1 and SMA2 as sources and dynamically changes the QL of any of them, no line inputs configured, the synce4l manages only local sources and synchronizes to best available QL source

Reply:
The DCR requirement was for system with ports. from the reason specified there it is clear that if there is no port, then no need to change QL while Synce running.
BTW, old external mode didnt support multiple SMAs...

Reply:
the implementation shall work with all the possible configurations, otherwise this is a bug, changing a QL dynamically is a basic concept that this PR introduces, same as getting a QL of a source which drives the dpll, already found which configs are not supported, please fix.

Reply:
fixed.
checking for best source. if port is best, then get from its rx, if ext_src is best, return its params.

I left the get_ql/ext_ql of tx in the code although no one uses them now. I dont think extra get functions are bad, but I can remove them if needed